src/Corax/Queries/SortingMatches/SortingMatch.BasicsComparers.cs

```diff
@@ -0,0 +1,381 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Corax.Queries.SortingMatches;
+
+internal static class BasicComparers
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static int CompareAscending(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
+    {
+        return x.SequenceCompareTo(y);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsSpace(byte c)
+    {
+        return c == ' ';
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsSpace(char c)
+    {
+        return c == ' ';
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsDigit(byte c)
+    {
+        return c >= '0' && c <= '9';
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsDigit(char c)
+    {
+        return c >= '0' && c <= '9';
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static byte ToUpper(byte c)
+    {
+        if (c >= 'a' && c <= 'z')
```

Comment:
For reference, for the analyzers I am building optimized vector aware version of many of the routines that are implemented here.

Reply:
So after both PRa are merged you we could use the optimized versions instead of those, right?